### PR TITLE
Rivets.js implementation

### PR DESCRIPTION
**Pull request by altmind**
This is new implementaion using rivetsjs https://github.com/mikeric/rivets

Rivets is tiny two-way binding library that is not opinionated and works well with all types of routers or models. It can also be considered "MVC" as it handles both changes on models, translates models to ui views and handle manual events.

I would say, that rivets is VERY similar to angularjs - they both are 2-way binding frameworks, they both use DOM-based views, they both define user events and bind model to dom in templates, they both have some sort of components(they are called binders in rivets) and filters(they are called formatters in rivets) and they both have the same data and behavior in controllers. Yey, I say that rivets is much simplier, less opinionated and much more composable(compatible with other frameworks).

Rivets is quite popular(2100 stars) and our company had enough experience to say that rivets is very mature. There was also a request for rivets #337 (sorry, this implementation does not use backbone and requirejs)

This implementation is using new UI (and thats why 'Clear completed button' tests fail)
This implementation also depends on director for rouring and jquery for dom manipulations.

Tests output: https://gist.github.com/altmind/16a38c0f79c2aceb89c8 (again, some tests fail because of breaking changes in new UI: #1110)

### CODE REVIEW

**Comment by altmind** [comment]
Can I help you somehow with integration of this example. Maybe answer some questions or give some examples?

**Comment by addyosmani** [comment]
A quick note: we're currently focused on #1110 (thanks for adding support for the new UI!), however definitely prefer this PR to one which integrates numerous other libraries together (Backbone + Rivets for example).  

As the team are still somewhat new to Rivets, we'll need to find time to grok and properly review this implementation.

**Comment by addyosmani** [comment]
@mikeric is there someone you could recommend who might be able to assist with the review of Rivets usage here? We would like to ensure it follows what your community consider best practices.

**Comment by mikeric** [comment]
@addyosmani, @altmind Ok, reviewing this now. Will make sure it follows Rivets best practices / patterns.

File: examples/rivets/readme.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+## Learning Rivets.js
+The [Rivets.js website](http://rivetsjs.com) is a great resource for getting started.
+
+Here are some links you may find helpful:
+
+* [Guide](http://rivetsjs.com/docs/guide/)
+* [Wiki](https://github.com/mikeric/rivets/wiki)
+
+Get help from other AngularJS users:
```
Comment (arthurvr) [inline examples/rivets/readme.md]: Angular? :D

File: examples/rivets/readme.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+* [rivets on Stack Overflow](http://stackoverflow.com/questions/tagged/rivets.js)
```
Comment (arthurvr) [inline examples/rivets/readme.md]: Rivets with a capital R.

**Comment by arthurvr** [comment]
Could you make sure the code passes jshint and jscs? Also, you're missing an EOL here and there.

**Comment by arthurvr** [comment]
Don't forget to add rivets to the home page as well.

File: examples/rivets/index.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+				</footer> 
+		</section>
+		<footer id="info">
+			<p>Double-click to edit a todo</p>
+			<p>Written by <a href="http://altio.us">Andrew Gurinovich</a></p>
+			<p>Part of <a href="http://todomvc.com">TodoMVC</a></p>
+		</footer>
+
+		<script src="node_modules/todomvc-common/base.js"></script>
+		<script src="node_modules/jquery/dist/jquery.min.js"></script>
```
Comment (arthurvr) [inline examples/rivets/index.html]: We use unminified vendor libs. https://github.com/tastejs/todomvc/issues/378

File: examples/rivets/.gitignore [hunk 4]
```diff
@@ -0,0 +1,23 @@
+
```
Comment (arthurvr) [inline examples/rivets/.gitignore]: Remove this line :)

File: examples/rivets/js/formatters.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                return todo.completed;
+            } else if (arg == '') {
+                return true;
+            }
+            return false;
+        });
+    };
+    
+})(rivets);
+
```
Comment (arthurvr) [inline examples/rivets/js/formatters.js]: Remove this line.

File: examples/rivets/.gitignore [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+node_modules/director/build/*
+!node_modules/director/build/director.js
+
+node_modules/todomvc-app-css/*
+!node_modules/todomvc-app-css/index.css
+
+node_modules/todomvc-common/*
+!node_modules/todomvc-common/base.css
+!node_modules/todomvc-common/base.js
+
```
Comment (arthurvr) [inline examples/rivets/.gitignore]: Remove this line.

**Comment by arthurvr** [comment]
Thanks for working on this @altmind. Sorry for the nitpicking! Punting the real review to @mikeric.

**Comment by mikeric** [comment]
@altmind The templates and data model in the app was spot on, but I rewrote things to make use of components instead of a single all-encompassing app object. I also replaced the domain-specific binders with generic/reusable ones.

That said, Rivets is pretty narrow in scope anyway (it only takes care of the view/UI part), so there really is no "Rivets way" of architecting an application. The adjustments I made are just so that it showcases components a bit and how you might start at writing a larger application using Rivets.

@arthurvr, @addyosmani I've opened a new pull request (branched off of this one) here: #1224. Let me know if anything else needs changing.

**Comment by arthurvr** [comment]
Closing in favor of https://github.com/tastejs/todomvc/pull/1224.

**Pull request closed without merging**